File: 1147-longest-chunked-palindrome-decomposition/1147-longest-chunked-palindrome-decomposition.js
/**
 * @param {string} text
 * @return {number}
 */
var longestDecomposition = function(s) {
    let res = 0; let n = s.length; let l = ""; let r = "";
    for(let i = 0; i < n; ++i) {
        l += s[i]; r = s[n - i - 1] + r;
        if(l === r) {
            res += 1;
            l = ""; r = "";
        }
    }
    return res;
};

File: 1147-longest-chunked-palindrome-decomposition/1147-longest-chunked-palindrome-decomposition.java
class Solution {
    public int longestDecomposition(String s) {
        int res = 0; int n = s.length(); String l = ""; String r = "";
        for(int i = 0; i < n; ++i) {
            l += s.charAt(i); r = s.charAt(n - i - 1) + r;
            if(l.equals(r)) {
                ++res;
                l = "";
                r = "";
            }
        }
        return res;
    }
}